spi/src/main/java/org/jboss/pnc/spi/datastore/predicates/ArtifactPredicates.java

```diff
@@ -65,8 +70,16 @@ public static Predicate<Artifact> withDependantBuildRecordId(Base32LongID buildR
 
     public static Predicate<Artifact> withDeliveredInProductMilestone(Integer productMilestoneId) {
         return (root, query, cb) -> {
-            Join<Artifact, ProductMilestone> productMilestones = root.join(Artifact_.deliveredInProductMilestones);
-            return cb.equal(productMilestones.get(ProductMilestone_.id), productMilestoneId);
+            Root<DeliverableArtifact> deliverableArtifact = query.from(DeliverableArtifact.class);
+            Join<DeliverableAnalyzerReport, DeliverableAnalyzerOperation> operation = deliverableArtifact
+                    .join(DeliverableArtifact_.report)
+                    .join(DeliverableAnalyzerReport_.operation);
+
+            return cb.and(
+                    cb.equal(operation.get(DeliverableAnalyzerOperation_.productMilestone), productMilestoneId),
+                    cb.equal(
+                            deliverableArtifact.get(DeliverableArtifact_.artifact).get(Artifact_.id),
+                            root.get(Artifact_.id)));
```

Comment:
Why is this condition needed?

Reply:
It's to bind together the two tables. This is not actually a join (because I think we can't make a join in this case) but 2 from tables.